src/bika/lims/browser/fields/uidreferencefield.py

```diff
@@ -155,18 +155,39 @@ def get(self, context, **kwargs):
         :return: object or list of objects for multiValued fields.
         :rtype: BaseContent | list[BaseContent]
         """
-        value = StringField.get(self, context, **kwargs)
-        if not value:
-            return [] if self.multiValued else None
+        uids = StringField.get(self, context, **kwargs)
+        if not self.multiValued:
+            uids = [uids]
+
+        # Remove non-valid uids
+        uids = filter(api.is_uid, uids)
+        if not uids:
+            if self.multiValued:
+                return []
+            return None
```

Comment:
not sure if this is required here. Maybe the UID catalog handles non-UIDs gracefully.
Therefore, I would move this check after the catalog query to avoid the possible IndexError at the end.

Reply:
You're right:

```
(Pdb++) uids
['2e91f8f02c6e4158a5bb45a40893e22c']
(Pdb++) uc(UID=uids)
[<Products.Archetypes.UIDCatalog.plugbrains object at 0x7fab109ad730>]
(Pdb++) uc(UID=[])
[]
(Pdb++) uc(UID=None)
[]
(Pdb++) uc(UID='')
[]
(Pdb++) uc(UID=[None, 'None'])
[]
(Pdb++) uc(UID=['', 'sa'])
[]
(Pdb++) len(uc())
0
```

Resolved with https://github.com/senaite/senaite.core/pull/2212/commits/ba548331aedb5334ce89c4d82ac69ab154dad522